hangman/lib/hangman/game.ex

```diff
@@ -0,0 +1,113 @@
+defmodule Hangman.Game do
+    
+  def new_game() do
+    %Hangman.Tally{}
+      |> Map.put(:letters, get_word())
+  end
+
+  defp get_word() do
+    Dictionary.random_word 
```

Comment:
This would be easier to explain face-to-face, so feel free to ask me to clarify this before or after class.
The trick of having "_x" change to "x" when guessed is interesting. If you'd implemented it differently, I'd give you extra credit for it. But, as it is, I think it makes the code harder to understand.

This approach introduces an abstraction: `letters` is a list that contains the letters in the word, where a letter is prepended with an underscore if it has not yet been guessed. So far, so good.

But once you introduce an abstraction, you need to wrap it in its own layer of code: a set of functions that let you make the queries you need on the internal form without knowing its implementation.

The alternative is to spread the knowledge of this representation throughout the rest of the code, often mixed into functions that do other things as well. This means that changing your representation becomes more difficult, and more error prone.

This probably seems like I'm being picky, but I think it's an important point to make on what is otherwise good code. 

### -3

Reply:
That is exactly what my code does and I totally agree with you on the fact that my `letters` abstraction makes my code harder to read.

Should I have wrapped the `letters` list in a separate module with its own functions that would manipulate the list? Also, if you are wrapping a data payload with a set of functions that allow the manipulation of the respective payload, would that be something similar with an object in an OOP paradigm?